evap/evaluation/models.py

```diff
@@ -283,17 +283,22 @@ def can_user_vote(self, user):
             and user in self.participants.all()
             and user not in self.voters.all())
 
+    def can_user_see_course(self, user):
+        if user.is_staff:
+            return True
+        if self.is_user_contributor_or_delegate(user):
+            return True
+        if self.is_private and user not in self.participants.all():
+            return False
+        return True
+
     def can_user_see_results(self, user):
         if user.is_staff:
             return True
         if self.state == 'published':
-            if self.is_user_contributor_or_delegate(user):
-                return True
             if not self.can_publish_grades:
                 return False
-            if self.is_private and user not in self.participants.all():
-                return False
-            return True
+            return self.can_user_see_course(user)
```

Comment:
this changes the methods behaviour. previously, contributors or delegates could always access the results, now only if course.can_publish_grades. i guess this is also the reason for the test failure.